crates/ika-core/src/dwallet_mpc/mod.rs

```diff
@@ -470,10 +461,10 @@ struct DeserializeMPCMessagesResponse<M: DeserializeOwned + Clone> {
 fn deserialize_mpc_messages<M: DeserializeOwned + Clone>(
     messages: Vec<HashMap<PartyID, MPCMessage>>,
 ) -> DeserializeMPCMessagesResponse<M> {
-    let mut deserialized_results = Vec::new();
+    let mut deserialized_results = HashMap::new();
     let mut malicious_parties = Vec::new();
```

Comment:
Its not a good idea to keep pushing into a vec as there could be repeated entries there.
```suggestion
    let mut malicious_parties = HashSet::new();
```

Reply:
Also, there could be a problem with ordering with Vec. This is why I always call my `deduplicate_and_sort()` when handling malicious parties. So you can keep as vec but call this method in the end if you prefer

Reply:
Please go over the code and call this method everywhere (you can do in separate pr) to assure we have no ordering issues with malicious parties reporting

Reply:
Added a todo and an issue